core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -251,6 +248,14 @@ protected[core] object WhiskWebActionsApi extends Directives {
         }
     }
 
+    private def headersFromJson(k: String, v: JsValue) : Seq[RawHeader] = v match {
+        case JsString(v)  => Seq(RawHeader(k, v))
+        case JsBoolean(v) => Seq(RawHeader(k, v.toString))
+        case JsNumber(v)  => Seq(RawHeader(k, v.toString))
+        case JsArray(v)   => v.map(headersFromJson(k, _:JsValue)).flatten
```

Comment:
Same comment on `flatMap`. In general, nesting a `_` can become tricky. I'd prefer fat-arrow notation in this case, the non inferred type seems confusing.

```scala
case JsArray(v)   => v.flatMap(inner => headersFromJson(k, inner))
```